desc/grid.py

```diff
@@ -447,31 +450,62 @@ def _create_nodes(
             if self.sym:
                 t += t[1] / 2
             dt = 2 * np.pi / t.size * np.ones_like(t)
+            # if endpoint:
+            #     FIXME: dt already has the correct weights
+            #         _scale_weights() function is about to mess it up
+            #         can test with endpoint = True, ntheta = 3
+
         else:
             t = np.atleast_1d(theta)
             dt = np.zeros_like(t)
             if t.size > 1:
-                dt[0] = (t[0] + t[1]) / 2
-                dt[1:-1] = (t[2:] - t[:-2]) / 2
-                dt[-1] = 2 * np.pi - (t[-2] + t[-1]) / 2
+                # choose dt to be half the cyclic distance of the surrounding two nodes
+                SUP = 2 * np.pi  # supremum
```

Comment:
Variable naming convention: is this all upper-case just because it's a hard-coded constant?

Reply:
Yes; if you prefer something else just let me know.